include/flamegpu/runtime/environment/DeviceEnvironmentDirectedGraph.cuh

```diff
@@ -0,0 +1,653 @@
+#ifndef INCLUDE_FLAMEGPU_RUNTIME_ENVIRONMENT_DEVICEENVIRONMENTDIRECTEDGRAPH_CUH_
+#define INCLUDE_FLAMEGPU_RUNTIME_ENVIRONMENT_DEVICEENVIRONMENTDIRECTEDGRAPH_CUH_
+
+#include "flamegpu/defines.h"
+#include "flamegpu/runtime/detail/curve/Curve.cuh"
+
+namespace flamegpu {
+/**
+ * Read-only accessor to a specific directed graph
+ * Directed graphs can only be modified via host functions
+ */
+class DeviceEnvironmentDirectedGraph {
+    const detail::curve::Curve::VariableHash graph_hash;
+
+ public:
+    /**
+    * This class is created when a vertex is provided to DeviceEnvironmentDirectedGraph::edgesLeaving()(id_t)
```

Comment:
```suggestion
    * This class is created when a vertex is provided to DeviceEnvironmentDirectedGraph::outEdges()(id_t)
```

Think this is the correct mapping? There are other occurences too that I might not have caught all of them in comments / docstrings.